iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -1306,10 +1367,22 @@ void LinalgBufferizePass::runOnFunction() {
         });
   };
 
-  auto walkResult = funcOp.walk([&](Operation *op) -> WalkResult {
-    b.setInsertionPoint(op);
-    return conversionDispatch(op);
+  // Forward init arguments from outer scf.for loop to the inner loops.
```

Comment:
If I understand this correctly, then this is just a cleanup . Does not actually affect the bufferization. If so, this might be better moved after the walk to do the conversion?

Reply:
No, this is actually need before the conversion to grantee tensors in the inner most loop aren't loop block arguments and they are something that is already can be mapped to a buffer.

Reply:
THis is true for a very specific use case that is coming from Linalg tiling. The loop block arguments must be explicitly mapped to buffers in the `convertSCFFor` operation. That is anyway modified to do `pre-order` walk. So the ops within the region will get the right tensor.

Reply:
The final conversion still `post-order` (I just made that explicit other than the default)

Reply:
that should be made pre-order. Would that work.

Reply:
No! the `pre-order` won't work with the way we do `convertAnyLinalgOp`. Because walking ops doesn't visit new ops added to the IR so we end up with a cloned op where the inner region didn't get converted and later deleted.

Reply:
yes, you should not clone the entire region. You should be creating a new `scf.for` operation, moving the body from the original operation into new operation and then the walk should take care of it?
The reason I am asking is when you change the result of an operation, you need to necessarily create a new op. Here it doesnt seem like you are needing to do that, which is strange. I'd rather know why

Reply:
There is no need to create a new `scf.for` all tensors uses and yielded have no use and they get canonicalized away (like other tensor ops in the region)

Reply:
Spoke offline about this. We shouldnt rely on this canonicalization pattern for the conversion and replacement here. There are some subtle semantic changes this entails (also I am fairly convinced that the canonicalization pattern removing the loop is wrong). Making the traversal pre-order would allow handling this correctly, but that requires `linalg` ops to be modified as well. This is a TODO (for me) after this CL lands.

Reply:
I am working of fixing the `PreOrder` walk issue. It is unrelated to this PR. So this is fine for now.